ledger/metrics/src/main/scala/com/daml/metrics/grpc/GrpcMetricsServerInterceptor.scala

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.metrics.grpc
+
+import com.daml.metrics.api.MetricHandle.Timer.TimerHandle
+import com.daml.metrics.api.MetricHandle.{Histogram, Meter, Timer}
+import com.daml.metrics.api.MetricsContext
+import com.daml.metrics.api.MetricsContext.{withExtraMetricLabels, withMetricLabels}
+import com.google.protobuf.{GeneratedMessage => JavaGeneratedMessage}
+import io.grpc.ForwardingServerCall.SimpleForwardingServerCall
+import io.grpc.ServerCall.Listener
+import io.grpc.{
+  ForwardingServerCallListener,
+  Metadata,
+  ServerCall,
+  ServerCallHandler,
+  ServerInterceptor,
+  Status,
+}
+import scalapb.{GeneratedMessage => ScalapbGeneratedMessage}
+
+class GrpcMetricsServerInterceptor(metrics: GrpcServerMetrics) extends ServerInterceptor {
+
+  override def interceptCall[ReqT, RespT](
+      serverCall: ServerCall[ReqT, RespT],
+      metadata: Metadata,
+      serverCallHandler: ServerCallHandler[ReqT, RespT],
+  ): ServerCall.Listener[ReqT] = {
+    val methodDescriptor = serverCall.getMethodDescriptor
+    val clientType = if (methodDescriptor.getType.clientSendsOneMessage()) "unary" else "streaming"
+    val serverType = if (methodDescriptor.getType.serverSendsOneMessage()) "unary" else "streaming"
+    withMetricLabels(
+      "grpc_service_name" -> methodDescriptor.getServiceName,
```

Comment:
Can we declare these as constants in the companion object, please? I.e., that way it'll be easier to reference them from the tests.